### PR TITLE
Finish memory implementation

### LINKED ISSUE #1: Finish memory implementation
There are steps to finish `mem` crate implementation:
- add +stable implementation of allocator implementation of `RawMem`
- add fuzzy tests for all implementations of `RawMem`
- review code and prove safe guarantees of this

### PR DESCRIPTION
**Pull request by konard**
Completes the mem crate implementation addressing all requirements from issue #1.

## Changes

- Added stable allocator-based `Alloc<T>` implementation
- Added comprehensive fuzzy tests for all `RawMem` implementations  
- Refactored test code using macros to eliminate repetition
- Moved tests to dedicated test files for better organization
- Cleaned up documentation and SAFETY comments per review feedback

## Testing

All tests passing:
- 14 fuzz tests covering invariants, cycles, edge cases, mutability, and stress tests
- 6 integration tests for Alloc type
- 4 passing doctests
- Tests verified on both Alloc and PreAlloc implementations

Fixes #1

### CODE REVIEW

File: crates/mem/src/alloc.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    let old_cap = self.cap;
+    let new_cap =
+      self.cap.checked_add(addition).ok_or(Error::CapacityOverflow)?;
+
+    // Check if new capacity exceeds layout limits
+    let layout =
+      Layout::array::<T>(new_cap).map_err(|_| Error::CapacityOverflow)?;
+
+    let ptr = if old_cap == 0 {
+      // Initial allocation
```
Comment (uselessgoddess) [inline crates/mem/src/alloc.rs]:
```suggestion
```

File: crates/mem/src/alloc.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+    if new_cap == 0 {
+      // Deallocate everything
+      if self.cap > 0 {
+        let layout =
+          Layout::array::<T>(self.cap).map_err(|_| Error::CapacityOverflow)?;
+
+        // SAFETY:
+        // - ptr was allocated with this layout
+        // - we're about to set cap to 0
```
Comment (uselessgoddess) [inline crates/mem/src/alloc.rs]: do not use lists notes in SAFETY comments

File: crates/mem/src/alloc.rs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+    // Shrink to new capacity
+    let old_layout =
+      Layout::array::<T>(self.cap).map_err(|_| Error::CapacityOverflow)?;
+    let new_layout =
+      Layout::array::<T>(new_cap).map_err(|_| Error::CapacityOverflow)?;
+
+    // SAFETY:
+    // - ptr was allocated with old_layout
+    // - new_layout.size() <= old_layout.size()
```
Comment (uselessgoddess) [inline crates/mem/src/alloc.rs]: avoid lists

**Review by uselessgoddess — CHANGES_REQUESTED** [review]

**Comment by konard** [comment]
## 🔄 Auto-restart 1/3

Detected uncommitted changes from previous run. Starting new session to review and commit them.

**Uncommitted files:**
```
?? issue-details.json
?? pr-details.json
?? review-comments.json
?? test-output.log
```

---
*Auto-restart will stop after changes are committed or after 2 more iterations. Please wait until working session will end and give your feedback.*
